_posts/2023-02-01-opentelemetry.md

```diff
@@ -0,0 +1,214 @@
+As already mentioned, the Jaeger backend can still be used together with the OTLP endpoint enabled, so let's use this one.
+
+#### Installing Jaeger backend
+
+The easiest way to deploy the Jaeger backend on Kubernetes is to use the Jaeger Operator.
+One of the pre-requisites for the latest releases is having the cert-manager already installed as well.
+
+The cert-manager installation is just about running the following command:
+
+```shell
+kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.6.3/cert-manager.yaml
+```
+
+After that, install the Jaeger Operator in the `observability` namespace:
+
+```shell
+kubectl create namespace observability
+kubectl create -f https://github.com/jaegertracing/jaeger-operator/releases/download/v1.41.0/jaeger-operator.yaml -n observability
+```
+
+With the Jaeger Operator up and running, you can now create a `Jaeger` custom resource describing the Jaeger instance to install.
+Following, the simplest `Jaeger` definition which is enough for our purposes:
+
+```yaml
+apiVersion: jaegertracing.io/v1
+kind: Jaeger
+metadata:
+  name: simplest
+```
+
+Create a `simplest.yaml` file with the above declaration and install it:
+
+```shell
+kubectl apply -f simplest.yaml
+```
+
+With the Jaeger instance running, an OTLP endpoint for getting the traces is exposed through the `http://simplest-collector:4317` service.
+
+> If using Minikube, you can use the port-forwarding to have access to the Jaeger Web UI easily from your local browser at `http://localhost:16686`. Just use the corresponding command `kubectl port-forward svc/simplest-query 16686:16686`.
+
+For more details, the official Jaeger documentation explains all the steps [here](https://www.jaegertracing.io/docs/latest/operator/).
+
+## Let's trace the HTTP bridge traffic
+
+Let's take the Strimzi HTTP bridge as an example of tracing to show how OpenTelemetry works.
+The first step is to deploy a `KafkaBridge` instance with the tracing enabled and the corresponding environment variables to configure the OTLP exporter.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaBridge
+metadata:
+  name: my-bridge
+spec:
+  replicas: 1
+  bootstrapServers: my-cluster-kafka-bootstrap:9092
+  http:
+    port: 8080
+  template:
+    bridgeContainer:
+      env:
+        - name: OTEL_SERVICE_NAME
+          value: my-otel-service
+        - name: OTEL_EXPORTER_OTLP_ENDPOINT
+          value: http://simplest-collector:4317
+  tracing:
+    type: opentelemetry
+```
+
+With the HTTP bridge up and running, we can use any HTTP client for sending and receiving messages.
+It could be a simple command line tool like `cURL`, a more UI oriented like [Postman](https://www.postman.com/) or whatever is your preferred one.
+You could also use an application you developed with your preferred language.
+For the purpose of this blog post, I used Postman by having a producer sending a couple of messages and the consumer getting them.
```

Comment:
I get it that you probably don't want to go through the details of how to use the bridge. But if nothing else, it would be great to link to the Bridge docs with how to do it.

But I think the ideal would be to use a client like the HTTP client from client examples for this:
* It would be easy to deploy with a simple command and make it easier for the user to see it.
* Right now, in all the pictures used below, it looks really weird with the traces only from the bridge. If I didn't know tracing already, I would probably be super confused about the value and what it actually shows. Having a proper client like we have in the examples would add there the sending and receiving messages and I think the full flow would make more sense and make it a bit clearer to the users what it does.

Reply:
> I get it that you probably don't want to go through the details of how to use the bridge. But if nothing else, it would be great to link to the Bridge docs with how to do it.

Yes exactly. Btw adding the link is a good idea.

> Having a proper client like we have in the examples would add there the sending and receiving messages and I think the full flow would make more sense and make it a bit clearer to the users what it does.

Do you mean the HTTP clients from the examples? In this case we have to add tracing support there because there is no such a support (no OpenTracing, no OpenTelemetry). We have it only for Kafka clients.

Reply:
I finally added the usage of the new Java SDK HTTP clients examples and corresponding new pictures.